tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py

```diff
@@ -0,0 +1,372 @@
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=self.pvc_size,
+            access_modes_rbd=[constants.ACCESS_MODE_RWO],
+            access_modes_cephfs=[constants.ACCESS_MODE_RWO],
+            sc_rbd=self.sc_obj,
+            project=self.proj_obj,
+        )
+
+    def test_pvc_expansion_when_full(self, threading_lock):
+        """
+        Verify PVC expansion when the encrypted PVC is 100% utilized.
+
+        Steps:
+            1. Create a encrypted storageclass.
+            2. Create a encrypted PVC of size 4G using defined vault configuration.
+            3. create a pod and mount the PVC inside the pod.
+            4. Fill FS 100% using FIO workload.
+            5. Expand PVC to size 10G
+            6. continue IO workload and verify the IOs are succeeding
+            7. Verify utilization alert will stop firing after volume expansion.
+
+        """
+        pvc_size_expanded = 10
+
+        # Run IO to utilise 100% of volume
+        log.info("Run IO on all to utilise 100% of PVCs")
+        for pod_obj in self.pods:
+            # for pod_obj in self.pod_objs:
+            pod_obj.run_io(
+                "fs",
+                bs="4k",
+                jobs=8,
+                size=f"{self.pvc_size}G",
+                io_direction="write",
+                fio_filename=f"{pod_obj.name}_f1",
+            )
+        log.info("Started IO on all to utilise 100% of PVCs")
+        # Wait for IO to finish
+        log.info("Wait for IO to finish on pods")
+        for pod_obj in self.pods:
+            # for pod_obj in self.pod_objs:
+            try:
+                pod_obj.get_fio_results()
+            except CommandFailed as cfe:
+                if "No space left on device" not in str(cfe):
+                    raise
+            log.info(f"IO finished on pod {pod_obj.name}")
+            # Verify used space on pod is 100%
+            used_space = get_used_space_on_mount_point(pod_obj)
+            assert used_space == "100%", (
+                f"The used space on pod {pod_obj.name} is not 100% " f"but {used_space}"
+            )
+            log.info(f"Verified: Used space on pod {pod_obj.name} is 100%")
+
+        prometheus_api = PrometheusAPI(threading_lock=threading_lock)
+
+        # Wait till utilization alerts starts
+        for response in TimeoutSampler(140, 5, prometheus_api.get, "alerts"):
+            alerts = response.json()["data"]["alerts"]
+            for pvc_obj in self.pvcs:
+                # for pvc_obj in self.pvc_objs:
+                alerts_pvc = [
+                    alert
+                    for alert in alerts
+                    if alert.get("labels", {}).get("persistentvolumeclaim")
+                    == pvc_obj.name
+                ]
+                # At least 2 alerts should be present
+                if len(alerts_pvc) < 2:
+                    break
+
+                # Verify 'PersistentVolumeUsageNearFull' alert is firing
```

Comment:
Is there a reason to check both PersistentVolumeUsageNearFull (75%) and PersistentVolumeUsageCritical (85%). Just trying to understand why use both as we try to exhaust the capacity, maybe checking 85% alert is enough?